src/audio/volume/volume.c

```diff
@@ -292,6 +292,27 @@ static void volume_ramp(struct comp_dev *dev)
 	vol_sync_host(dev, cd->channels);
 }
 
+static void reset_state(struct vol_data *cd)
+{
+	int i;
+
+	for (i = 0; i < PLATFORM_MAX_CHANNELS; i++) {
+		cd->volume[i]  =  MAX(MIN(cd->vol_max, VOL_ZERO_DB),
+				      cd->vol_min);
```

Comment:
I presume `VOL_ZERO_DB` cannot be outside of the `min-max` range?

Reply:
Nothing prevents topology to define e.g. volume scale from +10 dB to +50 dB. In that case volume would default to +10 dB until kernel sends a volume cmd(). Or if for some device safe (speaker with power limit) volume scale is -60 dB to -20 dB then volume would default to -20 dB that is within allowed range.

Reply:
ah, sure, you're right. So effectively this selects the closest to 0dB allowed value.